third_party/terraform/utils/error_retry_predicates.go

```diff
@@ -25,11 +30,34 @@ var defaultErrorRetryPredicates = []RetryErrorPredicateFunc{
 
 /** END GLOBAL ERROR RETRY PREDICATES HERE **/
 
-func isUrlTimeoutError(err error) (bool, string) {
-	if urlerr, ok := err.(*url.Error); ok && urlerr.Timeout() {
-		log.Printf("[DEBUG] Dismissed an error as retryable based on googleapis.com target: %s", err)
+type temporary interface {
+	Temporary() bool
+}
+
+func isTemporaryError(err error) (bool, string) {
+	if _, ok := err.(temporary); ok {
+		return true, "Got temporary error %v"
+	}
+	return false, ""
+}
+
+func isRetryableUrlError(err error) (bool, string) {
```

Comment:
I have a slight preference for implementing this as many smaller predicates, but it isn't too strongly held.

Reply:
:+1 am fine with that too. Thoughts on retrying on all net.OpErrors in general?

Reply:
I'm curious what the actual errors returned by that package end up looking like, if you think it would work I'm 👍. At the very least, errors for whom `Temporary()` and `Timeout()` return `true` make sense based on https://golang.org/src/net/net.go?s=13212:13333#L384.